test/bpf/Makefile

```diff
@@ -30,6 +30,10 @@ elf-demo.o: elf-demo.c
 	@$(ECHO_CC)
 	$(CLANG) ${FLAGS_CLANG} ${FLAGS} -I../../bpf/ $< -o $@
 
+nat-test: nat-test.c $(LIB)
+	@$(ECHO_CC)
+	$(CLANG) ${FLAGS} -I../../bpf/ -I./cmock  $< cmock/unity.c cmock/cmock.c ../../bpf/mock/mock_helpers.c ../../bpf/mock/mock_helpers_skb.c ../../bpf/mock/mock_conntrack_stub.c -o $@
```

Comment:
This builds the `nat-test` binary, but nothing is yet running the test.

There's already a `unit-test` target implied by lines 22 and 29, but could you maybe add one more small change to add this test to the regular suite?

One way to do this would be something like the following incremental patch:

```
$ git di
diff --git a/Makefile b/Makefile
index bffa7194572b..450233b82784 100644
--- a/Makefile
+++ b/Makefile
@@ -199,7 +199,6 @@ init-coverage: ## Initialize converage report for Cilium unit-tests.
 unit-tests: start-kvstores ## Runs all unit-tests for Cilium.
        $(QUIET) $(MAKE) $(SUBMAKEOPTS) -C tools/maptool/
        $(QUIET) $(MAKE) $(SUBMAKEOPTS) -C test/bpf/
-       test/bpf/unit-test
 ifeq ($(SKIP_VET),"false")
        $(MAKE) govet
 endif
diff --git a/test/bpf/Makefile b/test/bpf/Makefile
index f49afbc081aa..fadc02c85b1f 100644
--- a/test/bpf/Makefile
+++ b/test/bpf/Makefile
@@ -19,8 +19,10 @@ CLANG ?= $(QUIET) clang
 LLC ?= llc
 
 BPF_TARGETS := elf-demo.o
-TARGETS := $(BPF_TARGETS) unit-test
-all: $(TARGETS)
+ALL_TESTS := unit-test #nat-test
+TARGETS := $(BPF_TARGETS) $(ALL_TESTS)
+
+all: $(TARGETS) unit-tests
 
 elf-demo.o: elf-demo.c
        @$(ECHO_CC)
@@ -34,6 +36,12 @@ nat-test: nat-test.c $(LIB)
        @$(ECHO_CC)
        $(CLANG) ${FLAGS} -I../../bpf/ -I./cmock  $< cmock/unity.c cmock/cmock.c ../../bpf/mock/mock_helpers.c ../../bpf/mock/mock_helpers_skb.c ../../bpf/mock/mock_conntrack_stub.c -o $@
 
+unit-tests: $(ALL_TESTS)
+       @$(ECHO_CHECK)
+       $(foreach test,$<,$(ROOT_DIR)/$(RELATIVE_DIR)/$(test))
+
 clean:
        @$(ECHO_CLEAN)
        -$(QUIET)rm -f $(TARGETS)
+
+.PHONY: all unit-tests clean
```

I commented out unit-test in `$ALL_TESTS` above since it won't compile for me. Probably just need to replace the `../..` relative paths in the above command with instead `$(ROOT_DIR)`.

With this change, you can try it out also and confirm that this will be run from the CI by running `make -C test/bpf` and confirming that all the bpf unit tests build and run.

Reply:
Revised as you suggest!